problem_three.rb

```diff
@@ -0,0 +1,19 @@
+# Given an array of numbers having size N in which every number is between 1 and N, determine if there are any duplicates in it. 
+# - Do not sort the array or use bit vectors.
+# - Do not use 2 nesting for loops since the time complexity would be high.
+# - You are allowed to destroy the array if you like.
+# - Try to work within the array without using any other temporary data structures.
+# - Try to find a solution by executing just one loop.
+
+class ProblemThree
+  def duplicates?(array)
+    array.each_with_index do |_, index|
+      raise "array#{[index - 1]}=#{array[index - 1]} outside 1 and #{array.length}" \
```

Comment:
Why do you need this guard in place?

Reply:
If the current value in the loop is out of bounds, the algorithm will fail, therefore it makes sense to stop further looping and point the developer to the error.

Reply:
Given an array of numbers having size N in which **every number is between 1 and N**, determine if there are any duplicates in it. You are allowed to destroy the array if you like. Do not sort the array or use bit vectors. Try to work within the array without using any other temporary data structures. Do not use 2 nesting for loops since the time complexity would be high. Try to find a solution by executing just one loop.

Reply:
Removed redundant guard: https://github.com/Agilefreaks/Test/pull/42/commits/9b512f898bb4faa99d019d4f7d966489d0cdb9c4